packages/nodes-base/nodes/Totp/Totp.node.ts

```diff
@@ -0,0 +1,81 @@
+import type {
+	IExecuteFunctions,
+	INodeExecutionData,
+	INodeType,
+	INodeTypeDescription,
+} from 'n8n-workflow';
+
+import OTPAuth from 'otpauth';
+
+export class Totp implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'TOTP',
+		name: 'totp',
+		icon: 'fa:key',
+		group: ['transform'],
+		version: 1,
+		subtitle: '={{ $parameter["operation"] }}',
+		description: 'Generate a TOTP code',
+		defaults: {
+			name: 'TOTP',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'totpApi',
+				required: true,
+			},
+		],
+		properties: [
+			{
+				displayName: 'Operation',
+				name: 'operation',
+				type: 'options',
+				noDataExpression: true,
+				options: [
+					{
+						name: 'Generate Secret',
+						value: 'generateSecret',
+						action: 'Generate secret',
+					},
+				],
+				default: 'generateSecret',
+			},
+		],
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: INodeExecutionData[] = [];
+
+		const operation = this.getNodeParameter('operation', 0);
+		const credentials = (await this.getCredentials('totpApi')) as { label: string; secret: string };
+
+		const totp = new OTPAuth.TOTP({
+			issuer: 'GitHub',
```

Comment:
should `issuer` be a property on the credential as well?

Reply:
Yes, discussed in channel